main.cpp

```diff
@@ -77,22 +77,38 @@ void scan_demo(WiFiInterface *wifi)
 void http_demo(NetworkInterface *net)
 {
     TCPSocket socket;
+    nsapi_error_t response;
 
     printf("Sending HTTP request to www.arm.com...\r\n");
 
     // Open a socket on the network interface, and create a TCP connection to www.arm.com
     socket.open(net);
-    socket.connect("www.arm.com", 80);
+    response = socket.connect("www.arm.com", 80);
+    if(0 != response) {
```

Comment:
One question ,as we had this in our tests as well, should we try to connect multiple times or once is enough?